### PR TITLE
suppress `DatabaseError` correctly when generating fake data

### LINKED ISSUE #64: Fake data migration fails sometimes because of constraints in models
Known cases, but there may be more:

- `UNIQUE constraint failed: api_group.telegram_chat_url`
- `CHECK constraint failed: at_least_one_day_time_slot_must_be_selected`

### PR DESCRIPTION
**Pull request by lemontree210**
closes #64

### CODE REVIEW

File: api/migrations/0003_fake_data.py [hunk 1]
```diff
@@ -278,11 +278,16 @@ def _make_amount_of_recipe_and_skip_database_error(
...
-                    break
-                except DatabaseError:
-                    pass
+                    # context manager has to be used to avoid TransactionManagementError.
+                    # See section "Controlling transactions explicitly" in
+                    # https://docs.djangoproject.com/en/dev/topics/db/transactions/
+                    with transaction.atomic():
+                        recipe.make()
+                        break
+                except IntegrityError as e:
```
Comment (Roxe322) [inline api/migrations/0003_fake_data.py:289]: IntegrityError is a subclass of DatabaseError, so I think in previous version it also should work 🤔

Comment (lemontree210) [reply api/migrations/0003_fake_data.py:289]: OK, but the main thing is that I added `transaction.atomic()`

Comment (Roxe322) [reply api/migrations/0003_fake_data.py:289]: Ah, ok. But probably let's keep DatabaseError to avoid any other db exception?

Comment (lemontree210) [reply api/migrations/0003_fake_data.py]: Done

**Review by Roxe322 — APPROVED** [review]

**Pull request merged**